nodejs/week2/app.js

```diff
@@ -0,0 +1,93 @@
+import express from "express";
+import fs from "fs";
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+//Support parsing JSON requests
+app.use(express.json());
+
+let documents;
+try {
+  const data = fs.readFileSync("documents.json");
+  documents = JSON.parse(data);
+} catch (error) {
+  console.error("Error reading documents.json:", error);
+  documents = [];
+}
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+//Get /search
+app.get("/search", (req, res) => {
+  const query = req.query.q;
+
+  //If no query parameter is provided, return all documents
+  if (!query) {
+    return res.json(documents);
+  }
+
+  //Filter documents based on query parameter q
+  const result = documents.filter((doc) =>
+    Object.values(doc).some((value) =>
+      String(value).toLowerCase().includes(query.toLowerCase())
+    )
+  );
```

Comment:
Good job! Instead of search all the fields of a `doc` I would only extract the required fields. what if you documents library grows in attributes for each doc and suddenly your search engine searches in fields you dont want to. This is a question about being implicit or explicit. I like it to be more explicit. In this way you will also be sure that `value` is actually a string, which means you won't have to cast the type `String(...)`